### ISSUE TITLE
[BUG] eth_getBlockReceipts on block 98182138 returns invalid evm tx hash

### ISSUE THREAD

**Issue by quick-pawiromitchel** [sei-protocol/sei-chain#1833]
**Seid version**
```
name: sei
server_name: <appd>
version: v5.7.6
commit: 68597936bc3decc1c3f0495f05fbc8203bbe2cfa
build_tags: netgo,
go: go version go1.21.10 linux/amd64
build_deps:
- cosmossdk.io/errors@v1.0.0
- filippo.io/edwards25519@v1.0.0-rc.1
- github.com/99designs/keyring@v1.2.1
```

**Chain ID**
Sei Pacific - 1329

**Describe the bug**
Running `eth_getBlockReceipts` on block `98182138` returns invalid EVM tx hash `0x20060de0db9b36ccc8006267b254ea4526739bd95493c51fe3a82952fdbb4cb0`

**To Reproduce**
`eth_getBlockByNumber` doesn't return the tx hash
```
$ curl -s <HOST>   -X POST   -H "Content-Type: application/json"   --data '{"method":"eth_getBlockByNumber"
,"params":["0x5da23fa", true],"id":1,"jsonrpc":"2.0"}' | jq '.result.transactions[].hash'
"0x69d33816d036c61bf166d4641482631bde91a8d1c10417197fd262736f410f64"
"0xd87e1c1ab7053425adb3e847ca458b55c83d1f129559958fa4cbd4f2e1b99e7c"
"0xe801c21db88c7aa40e2808627c296b73e59541e5ea37de3665ead5ff8088b350"
"0xc1a52b1d326d48b0bed2fbb7ba237fe3a0487c566a500791d1eccda30be67ea2"
"0xe55adc64ce79a391f63f35620daa96c9082a70d1ea155962fe6853922aa969e6"
```
`eth_getBlockReceipts` does
```
$ curl -s <HOST>   -X POST   -H "Content-Type: application/json"   --data '{"method":"eth_getBlockReceipts","params":["0x5da23fa"],"id":1,"jsonrpc":"2.0"}' | jq '.result[].transactionHash'
"0x69d33816d036c61bf166d4641482631bde91a8d1c10417197fd262736f410f64"
"0xd87e1c1ab7053425adb3e847ca458b55c83d1f129559958fa4cbd4f2e1b99e7c"
"0xe801c21db88c7aa40e2808627c296b73e59541e5ea37de3665ead5ff8088b350"
"0xc1a52b1d326d48b0bed2fbb7ba237fe3a0487c566a500791d1eccda30be67ea2"
"0x20060de0db9b36ccc8006267b254ea4526739bd95493c51fe3a82952fdbb4cb0"
"0xe55adc64ce79a391f63f35620daa96c9082a70d1ea155962fe6853922aa969e6"
```

Trying to lookup information about that hash returns `null`
```
curl -s <HOST>  -X POST   -H "Content-Type: application/json"   --data '{"method":"eth_getTransactionByHash","params":["0x20060de0db9b36ccc8006267b254ea4526739bd95493c51fe3a82952fdbb4cb0"],"id":1,"jsonrpc":"2.0"}'
```
`{"jsonrpc":"2.0","id":1,"result":null}`

**Expected behavior**
`0x20060de0db9b36ccc8006267b254ea4526739bd95493c51fe3a82952fdbb4cb0` should not be in the result of `eth_getBlockReceipts`. 

**Additional context**
The transaction looks to be a non-evm tx ], removing the `0x` from the hash, it shows up on the explorer https://seistream.app/transactions/20060DE0DB9B36CCC8006267B254EA4526739BD95493C51FE3A82952FDBB4CB0

**Comment by philipsu522** [comment]
hey @quick-pawiromitchel this should be fixed in the next release!

**Comment by lordshisho** [comment]
This is intended behavior.

https://v2.seipex.fi/tx?hash=0x20060DE0DB9B36CCC8006267B254EA4526739BD95493C51FE3A82952FDBB4CB0

It's a Cosmos tx that interacted with the EVM and produced logs under the Cosmos tx hash.

https://v2.seipex.fi/block?hash=0xea7163ad7cae2ec05997bb5e5b3bed225de6dd815d1bb11d318f731a218c5297

You won't find it with `eth_getBlockByHash` but you will with `eth_getBlockReceipts`.

https://v2.seipex.fi/blockreceipts?hash=0xea7163ad7cae2ec05997bb5e5b3bed225de6dd815d1bb11d318f731a218c5297

Long story short, this is a valid tx hash that had a valid state transition and emit logs on the EVM.

**Comment by cordt-sei** [comment]
As mentioned this is (was) intended. These have been removed, and are now only seen through the custom `sei_*` methods.
